handlers/issue.go

```diff
@@ -1,71 +1,79 @@
 package handlers
 
 import (
+	"context"
 	"encoding/json"
-	"fmt"
-	"io/ioutil"
-	"net/http"
 	"regexp"
 	"strings"
 
 	"github.com/golang/glog"
 	"github.com/google/go-github/github"
 )
 
+const (
+	Assign   = "/assign"
+	Unassign = "/unassign"
+)
+
 type GithubIssue github.Issue
 
 func (s *Server) handleIssueEvent(body []byte) {
-	glog.Infof("Received an Issue Event")
+	glog.Info("Received an Issue Event")
 
 }
 
-func (s *Server) handleIssueCommentEvent(body []byte) {
-	glog.Infof("Received an IssueComment Event")
+//function to handle issue comments
+func (s *Server) handleIssueCommentEvent(body []byte, client *github.Client) {
+	var commentEvent github.IssueCommentEvent
 
-	var prc github.IssueCommentEvent
-	err := json.Unmarshal(body, &prc)
+	err := json.Unmarshal(body, &commentEvent)
 	if err != nil {
 		glog.Errorf("fail to unmarshal: %v", err)
 	}
-	glog.Infof("prc: %v", prc)
-	comment := *prc.Comment.Body
+	//Get the owner info from the req body, and use this owner for the all api's
+	owner := *commentEvent.Repo.Owner.Login
+	ctx := context.Background()
+	//extract the user from login to validate whether user is collaborator or not
+	user := commentEvent.Sender.Login
+	comment := *commentEvent.Comment.Body
+	//split the assignees and operation to be performed.
+	substrings := strings.Split(comment, "@")
 
-	// https://github.com/islinwb/test/pull/1
-	prID := strings.SplitAfter(prc.Issue.PullRequestLinks.GetHTMLURL(), "github.com/")[1]
-	// https://github.com/islinwb/test/pull/1.patch
-	// From <commit ID> MON ...
-	patchURL := prc.Issue.PullRequestLinks.GetPatchURL()
-	resp, err := http.Get(patchURL)
+	assign, _, err := client.Repositories.IsCollaborator(ctx, owner, *commentEvent.Repo.Name, *user)
 	if err != nil {
-		fmt.Println(err)
-	}
+		glog.Fatalf("Not the collaborator: %v", err)
```

Comment:
The definition of `IsCollaborator` is:

```
// IsCollaborator checks whether the specified GitHub user has collaborator
// access to the given repo.
// Note: This will return false if the user is not a collaborator OR the user
// is not a GitHub user.
//
// GitHub API docs: https://developer.github.com/v3/repos/collaborators/#get
func (s *RepositoriesService) IsCollaborator(ctx context.Context, owner, repo, user string) (bool, *Response, error) {
	u := fmt.Sprintf("repos/%v/%v/collaborators/%v", owner, repo, user)
	req, err := s.client.NewRequest("GET", u, nil)
	if err != nil {
		return false, nil, err
	}

	resp, err := s.client.Do(ctx, req, nil)
	isCollab, err := parseBoolResponse(err)
	return isCollab, resp, err
}
```

So, there are two scenarios where `IsCollaborator` return an error. You are logging the error message "Not the collaborator: err", I just feel confused about the information. Do you want to say, 'foo' is not a valid github ID or a repo collaborator?

Reply:
When an error occurs, any reason why we continue the process?

Reply:
@m1093782566 , As we run the comment handle in go routine, we can exit the function, it must be continued to handle the further requests.

Reply:
OK, I think I just missed that it's fatalf(). So, please improve the error message.

Reply:
> The definition of `IsCollaborator` is:
> 
> ```
> // IsCollaborator checks whether the specified GitHub user has collaborator
> // access to the given repo.
> // Note: This will return false if the user is not a collaborator OR the user
> // is not a GitHub user.
> //
> // GitHub API docs: https://developer.github.com/v3/repos/collaborators/#get
> func (s *RepositoriesService) IsCollaborator(ctx context.Context, owner, repo, user string) (bool, *Response, error) {
> 	u := fmt.Sprintf("repos/%v/%v/collaborators/%v", owner, repo, user)
> 	req, err := s.client.NewRequest("GET", u, nil)
> 	if err != nil {
> 		return false, nil, err
> 	}
> 
> 	resp, err := s.client.Do(ctx, req, nil)
> 	isCollab, err := parseBoolResponse(err)
> 	return isCollab, resp, err
> }
> ```
> 
> So, there are two scenarios where `IsCollaborator` return an error. You are logging the error message "Not the collaborator: err", I just feel confused about the information. Do you want to say, 'foo' is not a valid github ID or a repo collaborator?

`IsCollaborator` just checks the comment handler is a collaborator or not?
If not, we just log the error and proceed to handle other requests